fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -1924,6 +1924,7 @@ public void handleDMLStmt(ExecPlan execPlan, DmlStmt stmt) throws Exception {
             }
             throw new UserException(t.getMessage());
         } finally {
+            QeProcessorImpl.INSTANCE.unregisterQuery(context.getExecutionId());
```

Comment:
This function is too complicated, better to be refactored later.
In some case, the query may not be registered. Is it a appropriate to unregister it?

Reply:
In other scenarios, there should be a try-finally to unregister the query